server/src/test/scala/controllers/UsersControllerSpec.scala

```diff
@@ -547,12 +549,64 @@ class UsersControllerSpec extends PlayPostgresSpec with LoginUtils {
 
       val verificationToken = UUID.randomUUID()
       when(tokenGenerator.next()).thenReturn(verificationToken)
-      client.createUser(userRequest).futureValue
-      
+
+      val userCreated = client.createUser(userRequest).futureValue
+
       val response = client.sendEmailVerificationToken(request).futureValue
-      response.expiresAt mustNot be (null)
+
+      val token = userTokensRepository
+        .find(userCreated.id)
+        .futureValue
+        .find(_.tokenType == UserTokenType.EmailVerification)
+        .value
+      
+      response.expiresAt must be (token.expiresAt)
     }
 
+    "success on verifying email and login" in withApiClient { client =>
+      val name = Name.trusted("wiringbits")
+      val email = Email.trusted("test1@email.com")
+      val captcha =  Captcha.trusted("test")
+      val password = Password.trusted("test123...")
+      val request = SendEmailVerificationToken.Request(
+        email = email,
+        captcha = captcha
+      )
+
+      val userRequest = CreateUser.Request(
+        name = name,
+        email = email,
+        password = password,
+        captcha = captcha
+      )
+
+      userRepository.find(email).value
+      val verificationToken = UUID.randomUUID()
+      when(tokenGenerator.next()).thenReturn(verificationToken)
+
+      val userCreated = client.createUser(userRequest).futureValue
+
+      val userTokenCreate = userTokensRepository
+        .find(userCreated.id)
+        .futureValue
+        .find(_.tokenType == UserTokenType.EmailVerification)
+        .value
+      userTokensRepository.delete(userTokenCreate.id, userCreated.id)
```

Comment:
`.futureValue` is likely missing, ideally, there wouldn't be a need to run this, if you have a conflict, just run `when(tokenGenerator.next()).thenReturn(verificationToken)` before the next call by providing a new uuid